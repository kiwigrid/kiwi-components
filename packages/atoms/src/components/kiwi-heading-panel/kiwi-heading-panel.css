kiwi-heading-panel {
  display: block;
}
