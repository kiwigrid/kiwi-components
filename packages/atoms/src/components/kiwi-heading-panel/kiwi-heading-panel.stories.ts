import { html, TemplateResult } from 'lit-html';
import notes from './readme.md';

export default {
  title: 'KiwiComponents/HeadingPanel',
  parameters: { docs: { description: { component: notes } } },
};

export const Basic = ({
  secondary,
  top,
  left,
  right,
}: Record<string, boolean>): TemplateResult => {
  console.log({
    secondary,
    top,
    left,
    right,
  });

  return html`
    <kiwi-heading-panel
      .withSecondary="${secondary}"
      .withTop="${top}"
      .withLeft="${left}"
      .withRight="${right}"
    >
      <span slot="kiwi-heading-top"> Breadcrumb » Goes » Here </span>

      <span
        slot="kiwi-heading-left"
        class="glyphicon glyphicon-flash icon-sm"
      ></span>

      <span slot="kiwi-heading-primary">Primary Heading</span>

      <span slot="kiwi-heading-secondary" class="text-default">
        Secondary Heading
      </span>

      <nav slot="kiwi-heading-right">
        <ul class="nav nav-pills">
          <li class="active">
            <a>Pill1</a>
          </li>
          <li>
            <a>Pill2</a>
          </li>
        </ul>
      </nav>
    </kiwi-heading-panel>
  `;
};
Basic.args = {
  secondary: true,
  top: true,
  left: true,
  right: true,
};
