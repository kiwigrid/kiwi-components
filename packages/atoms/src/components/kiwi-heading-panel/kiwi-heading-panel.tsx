import {
  Component,
  FunctionalComponent,
  h,
  Host,
  JSX,
  Prop,
} from '@stencil/core';

type Slots = {
  [Key in
    | 'primary'
    | 'secondary'
    | 'top'
    | 'left'
    | 'right']: `kiwi-heading-${Key}`;
};

const slots: Slots = {
  primary: 'kiwi-heading-primary',
  secondary: 'kiwi-heading-secondary',
  left: 'kiwi-heading-left',
  right: 'kiwi-heading-right',
  top: 'kiwi-heading-top',
};

/**
 * @slot primary - Content for primary heading.
 * @slot secondary - Optional secondary heading.
 * @slot top - Optional content to be displayed above the primary heading, e.g. a breadcrumb
 * @slot left - Optional content to be displayed left of the primary heading, e.g. an icon
 * @slot right - Optional content to be displayed right, e.g. navigation
 */
@Component({
  tag: 'kiwi-heading-panel',
  styleUrl: 'kiwi-heading-panel.css',
  shadow: false,
})
export class KiwiHeadingPanel {
  /** Show slot `kiwi-heading-secondary`. */
  @Prop()
  public withSecondary = false;

  /** Show slot `kiwi-heading-top`. */
  @Prop()
  public withTop = false;

  /** Show slot `kiwi-heading-left`. */
  @Prop()
  public withLeft = false;

  /** Show slot `kiwi-heading-right`. */
  @Prop()
  public withRight = false;

  render(): JSX.Element {
    return (
      <Host class="panel panel-default">
        <div class="panel-heading py-2" hidden={!this.withTop}>
          <KiwiHeadingSlot name="top" />
        </div>

        <div class="panel-body d-flex align-items-center">
          <KiwiHeadingSlot
            name="left"
            hidden={!this.withLeft}
            classes={{
              'align-self-start': this.withSecondary,
              'mt-1': this.withSecondary,
              'mr-2': true,
            }}
          />

          <h1 class="h2 m-0" style={{ flex: '1 0 auto' }}>
            <KiwiHeadingSlot name="primary" />
            <div class="mt-1" hidden={!this.withSecondary}>
              <small>
                <KiwiHeadingSlot name="secondary" />
              </small>
            </div>
          </h1>

          <KiwiHeadingSlot name="right" hidden={!this.withRight} />
        </div>
      </Host>
    );
  }
}

const KiwiHeadingSlot: FunctionalComponent<{
  name: keyof Slots;
  hidden?: boolean;
  classes?: string | Record<string, boolean>;
}> = ({ name, hidden, classes = {} }) => (
  <div hidden={hidden} key={name} class={classes}>
    <slot name={slots[name]} />
  </div>
);
