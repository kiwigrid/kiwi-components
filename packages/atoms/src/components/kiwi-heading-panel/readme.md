# kiwi-heading-panel



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description                         | Type      | Default |
| --------------- | ---------------- | ----------------------------------- | --------- | ------- |
| `withLeft`      | `with-left`      | Show slot `kiwi-heading-left`.      | `boolean` | `false` |
| `withRight`     | `with-right`     | Show slot `kiwi-heading-right`.     | `boolean` | `false` |
| `withSecondary` | `with-secondary` | Show slot `kiwi-heading-secondary`. | `boolean` | `false` |
| `withTop`       | `with-top`       | Show slot `kiwi-heading-top`.       | `boolean` | `false` |


## Slots

| Slot          | Description                                                                   |
| ------------- | ----------------------------------------------------------------------------- |
| `"left"`      | Optional content to be displayed left of the primary heading, e.g. an icon    |
| `"primary"`   | Content for primary heading.                                                  |
| `"right"`     | Optional content to be displayed right, e.g. navigation                       |
| `"secondary"` | Optional secondary heading.                                                   |
| `"top"`       | Optional content to be displayed above the primary heading, e.g. a breadcrumb |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
