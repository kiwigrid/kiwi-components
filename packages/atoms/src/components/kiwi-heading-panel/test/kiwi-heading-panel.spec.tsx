import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { KiwiHeadingPanel } from '../kiwi-heading-panel';

describe('kiwi-heading-panel', () => {
  it.each`
    top      | left     | right    | secondary
    ${false} | ${false} | ${false} | ${false}
    ${false} | ${false} | ${false} | ${true}
    ${false} | ${false} | ${true}  | ${false}
    ${false} | ${false} | ${true}  | ${true}
    ${false} | ${true}  | ${false} | ${false}
    ${false} | ${true}  | ${false} | ${true}
    ${false} | ${true}  | ${true}  | ${false}
    ${false} | ${true}  | ${true}  | ${true}
    ${true}  | ${false} | ${false} | ${false}
    ${true}  | ${false} | ${false} | ${true}
    ${true}  | ${false} | ${true}  | ${false}
    ${true}  | ${false} | ${true}  | ${true}
    ${true}  | ${true}  | ${false} | ${false}
    ${true}  | ${true}  | ${false} | ${true}
    ${true}  | ${true}  | ${true}  | ${false}
    ${true}  | ${true}  | ${true}  | ${true}
  `(
    'renders [top: $top], [left: $left], [right: $right],Â [secondary: $secondary]',
    async ({ top, right, left, secondary }: Record<string, boolean>) => {
      const page = await newSpecPage({
        components: [KiwiHeadingPanel],
        template: () => (
          <kiwi-heading-panel
            withTop={top}
            withLeft={left}
            withRight={right}
            withSecondary={secondary}
          >
            <span slot="kiwi-heading-primary">Primary</span>
            <span slot="kiwi-heading-top">Top</span>
            <span slot="kiwi-heading-left">Left</span>
            <span slot="kiwi-heading-right">Right</span>
            <span slot="kiwi-heading-secondary">Secondary</span>
          </kiwi-heading-panel>
        ),
      });

      expect(page.root).toMatchSnapshot();
    },
  );
});
