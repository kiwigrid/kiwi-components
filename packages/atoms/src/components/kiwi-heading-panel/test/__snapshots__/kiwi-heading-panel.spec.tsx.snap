// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-heading-panel renders [top: false], [left: false], [right: false], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: false], [left: false], [right: false], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: false], [left: false], [right: true], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: false], [left: false], [right: true], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: false], [left: true], [right: false], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: false], [left: true], [right: false], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: false], [left: true], [right: true], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: false], [left: true], [right: true], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2" hidden="">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: false], [right: false], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: false], [right: false], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: false], [right: true], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: false], [right: true], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1" hidden=""></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: true], [right: false], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: true], [right: false], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div hidden=""></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: true], [right: true], [secondary: false] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="mr-2"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1" hidden="">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;

exports[`kiwi-heading-panel renders [top: true], [left: true], [right: true], [secondary: true] 1`] = `
<kiwi-heading-panel class="panel panel-default">
  <span slot="kiwi-heading-primary">
    Primary
  </span>
  <span slot="kiwi-heading-top">
    Top
  </span>
  <span slot="kiwi-heading-left">
    Left
  </span>
  <span slot="kiwi-heading-right">
    Right
  </span>
  <span slot="kiwi-heading-secondary">
    Secondary
  </span>
  <div class="panel-heading py-2">
    <div></div>
  </div>
  <div class="align-items-center d-flex panel-body">
    <div class="align-self-start mr-2 mt-1"></div>
    <h1 class="h2 m-0" style="flex: 1 0 auto;">
      <div></div>
      <div class="mt-1">
        <small>
          <div></div>
        </small>
      </div>
    </h1>
    <div></div>
  </div>
</kiwi-heading-panel>
`;
